src/app/space/create/deployments/deployments-donut/deployments-donut.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { debounce } from 'lodash';
+
+@Component({
+  selector: 'deployments-donut',
+  templateUrl: './deployments-donut.component.html',
+  styleUrls: ['./deployments-donut.component.less']
+})
+export class DeploymentsDonutComponent implements OnInit {
+
+  @Input() mini: boolean;
+  @Input() applicationId: string;
+
+  isIdled = false;
+  scalable = true;
+  pods: any;
+  desiredReplicas: number;
+
+  private scaleRequestPending = false;
+  private debounceScale: () => void;
+
+  constructor() { }
+
+  ngOnInit(): void {
+    this.debounceScale = debounce(this.scale, 650);
+    this.pods = this.getPods();
+    this.desiredReplicas = this.getDesiredReplicas();
+  }
+
+  scaleUp(): void {
+    if (!this.scalable) {
+      return;
+    }
+    let desired = this.getDesiredReplicas();
+    this.desiredReplicas = desired + 1;
+
+    this.scaleRequestPending = true;
+    this.debounceScale();
+  }
+
+  scaleDown(): void {
+    if (!this.scalable) {
+      return;
+    }
+
+    if (0 === this.getDesiredReplicas()) {
+      return;
+    }
+
+    let desired = this.getDesiredReplicas();
+    this.desiredReplicas = desired - 1;
+
+    this.scaleRequestPending = true;
+    this.debounceScale();
+  }
+
+  getDesiredReplicas(): number {
+    if (this.desiredReplicas) {
+      return this.desiredReplicas;
+    }
+
+    // TODO: acquire replicas from service
```

Comment:
Up to you if you do this now or along with the future work to actually get values from the service, but functions like this are going to need to return Observables, and the components will either have to (un)subscribe on them or use the async pipe in templates. I think using Observables might change some logic like the guard at the top of this function.

Reply:
Yes I'd like to do this as a subsequent changeset. I actually expect quite a few logic changes throughout the code as even the format of `pods` isn't really set in stone yet.